src/datetimesettings.cpp

```diff
@@ -100,40 +150,64 @@ void DateTimeSettings::setTimezone(const QString &tz)
         return;
     }
 
-    m_time.setTimezone(tz);
+    Maemo::Timed::WallClock::Settings s;
+    s.setTimezoneManual(tz);
+    setSettings(s);
 }
 
 void DateTimeSettings::setHourMode(DateTimeSettings::HourMode mode)
 {
-    Maemo::Timed::Interface timedInterface;
-    Maemo::Timed::WallClock::Settings settings;
-    settings.setFlag24(mode == TwentyFourHours);
-    timedInterface.wall_clock_settings_async(settings);
+    Maemo::Timed::WallClock::Settings s;
+    s.setFlag24(mode == TwentyFourHours);
+    setSettings(s);
+}
+
+bool DateTimeSettings::setSettings(Maemo::Timed::WallClock::Settings &s)
+{
+    if (!s.check()) {
+        return false;
+    }
+
+    m_timed.wall_clock_settings_async(s);
```

Comment:
How about tracking the pending call/reply, and print an error msg in case of trouble, like DateTimeSettings::onGetWallClockInfoFinished?